### ISSUE TITLE
Make Kahoy git repository aware when loading manifests (git `storage.ResourceRepository`)

### ISSUE THREAD

**Issue by slok** [slok/kahoy#43]
## Context

At this moment we are loading the manifests from 2 different paths, planning the states...

We implement this with an `fs.Repository` type `storage.Repository` having an `old` and a `new` `storage.Repository`, with both we can get the plan of whats should exist and what should not exist.

On top of these, we have created a factory that knows how to return the `old` and `new` repos, this gives us the ability to be _smarter_ when creating both storages.

## Problem

The common case for `Kahoy` is to be used in _gitops_ style, so normally we will have the state on a Git repository, with the `paths` mode implementation, we can prepare the repositories previously ([e.g][script]). But this is messy, having bash scripts around to prepare the 

## Solution

We could implement a `git` mode that would make all these preparation inside Kahoy using the repository loaded in memory. We would load 2 times the fs repo, set the `old` in the `before-commit` state, and use these two repositories as a regular `fs.Repository` implementation but from memory, each one in its own git state (emulating two paths).

Basically what the [example script][script] does but from memory and inside Kahoy.

[script]: https://gist.github.com/slok/3f37c2a0dd823d5b66db869a468109ce
